src/util/filesystem.hxx

```diff
@@ -5,12 +5,67 @@
 
 #if __has_include(<filesystem>)
 #include <filesystem>
-namespace fs = std::filesystem;
 #elif __has_include(<experimental/filesystem>)
 #include <experimental/filesystem>
-namespace fs = std::experimental::filesystem;
 #else
 #pragma message("Warning: Your compiler does not support the filesystem library. Please report this incident to the Cytopia Team")
 #endif
 
+namespace fs
+{
+#if __has_include(<filesystem>)
+using namespace std::filesystem;
+#elif __has_include(<experimental/filesystem>)
+using namespace std::experimental::filesystem;
+#endif
+
+/** Read contents from a file as string.
+   * @oaram fileName Name of the file to read
+   * @param binaryMode (optional) open the file in binary mode. (default: false)
+   * @returns the content of the file as string
+   */
+std::string readFileAsString(const std::string &fileName, bool binaryMode = false);
```

Comment:
binary mode is needed for the compressed map. This needs to be refactored to export real binary data instead of a compressed json string anyway. i created #528 for that.

Let's leave it as is for now.